src/methods.jl

```diff
@@ -10,318 +19,453 @@ derivatives.
+        @doc """
+    $($(Meta.quot(fdm_fun)))(
+        p::Int,
+        q::Int;
+        adapt::Int=1,
+        condition::Int=DEFAULT_CONDITION,
+        geom::Bool=false
+    )
+
+Contruct a finite difference method at a $($(Meta.quot(direction))) grid of `p` linearly
+spaced points.
+
+# Arguments
+- `p::Int`: Number of grid points.
+- `q::Int`: Order of the derivative to estimate.
+
+# Keywords
+- `adapt::Int=1`: Use another finite difference method to estimate the magnitude of the
+    `p`th order derivative, which is important for the step size computation. Recurse
+    this procedure `adapt` times.
+- `condition::Int`: Condition number. See [`DEFAULT_CONDITION`](@ref).
+- `geom::Bool`: Use geometrically spaced points instead of linearly spaced points.
+
+# Returns
+- `FiniteDifferenceMethod`: The specified finite difference method.
+        """ $fdm_fun
+    end
 end
 
-# Handle inputs that aren't `AbstractFloat`s -- assume what you wanted was a `Float64`. This
-# most common way that this method gets invoked is when `x` is an `Integer`.
-function fdm(m::FiniteDifferenceMethod, f, x::Real, val; kwargs...)
-    return fdm(m, f, Float64(x), val; kwargs...)
+function _make_adaptive_bound_estimator(
+    constructor::Function,
+    q::Int,
+    adapt::Int,
+    condition::Int;
+    kw_args...
+)
+    if adapt >= 1
+        estimate_derivative =
+            constructor(q + 1, q, adapt=adapt - 1, condition=condition; kw_args...)
+        return (f, x) -> maximum(abs.(estimate_derivative(f, x)))
+    else
+        return _make_default_bound_estimator(condition=condition)
+    end
 end
 
-function fdm(m::FiniteDifferenceMethod, f, x::Real, ::Val{false}=Val(false); kwargs...)
-    _, dfdx = fdm(m, f, x, Val(true); kwargs...)
-    return dfdx
-end
+_forward_grid(p::Int) = 0:(p - 1)
 
+_backward_grid(p::Int) = (1 - p):0
 
-## Deprecations
+function _central_grid(p::Int)
+    if isodd(p)
+        return div(1 - p, 2):div(p - 1, 2)
+    else
+        return vcat(div(-p, 2):-1, 1:div(p, 2))
+    end
+end
 
-# Used for keyword argument name deprecations
-function _dep_kwarg(kwargs)
-    for (old, new) in [(:ε, :eps), (:M, :bound)]
-        haskey(kwargs, old) || continue
-        val = kwargs[old]
-        error(
-            "keyword argument `", old, "` should now be passed as `", new, "` upon ",
-            "application of the method. For example:\n    ",
-            "central_fdm(5, 1)(f, x; $new=$val)\n",
-            "not\n    ",
-            "central_fdm(5, 1; $old=$val)(f, x)"
-        )
+_exponentiate_grid(grid::Vector, base::Int=3) = sign.(grid) .* base .^ abs.(grid) ./ base
+
+function _is_symmetric(vec::Vector; centre_zero=false, negate_half=false)
+    n = div(length(vec), 2)
+    half_sign = negate_half ? -1 : 1
+    if isodd(length(vec))
+        centre_zero && vec[n + 1] != 0 && return false
+        return vec[1:n] == half_sign .* reverse(vec[n + 2:end])
+    else
+        return vec[1:n] == half_sign .* reverse(vec[n + 1:end])
```

Comment:
How do you feel about 
```suggestion
function _is_symmetric(vec::Vector; centre_zero=false, negate_half=false)
    half_sign = negate_half ? -1 : 1
    if isodd(length(vec))
        centre_zero && vec[end÷2 + 1] != 0 && return false
        return vec[1:end÷2] == half_sign .* reverse(vec[2+end÷2:end])
    else
        return vec[1:end÷2] == half_sign .* reverse(vec[1+end÷2:end])
```

I am not sure how i feel about it

Reply:
Also happy with that. Should there be spaces around the operators inside the `[...]`s?

Reply:
I am not sure, I would need to look at th style guide closely.

I am not sure if my code if more or less clear

Reply:
> spaces around the operators

style guide recommends to write this like e.g. `vec[(end ÷ 2 + 1):end])`

Reply:
Thanks, @nickrobinson251.